xpra/clipboard/clipboard_core.py

```diff
@@ -532,7 +532,16 @@ def remote_targets(self, local_targets:Iterable[str]):
         return _filter_targets(local_targets)
 
     def _munge_raw_selection_to_wire(self, target:str, dtype:str, dformat:int, data) -> Tuple[Any,Any]:
-        log("_munge_raw_selection_to_wire%s", (target, dtype, dformat, repr_ellipsized(bytestostr(data))))
+        log.warn("_munge_raw_selection_to_wire, target=%s, type=%s, format=%s, len(data)=%s",
+                 target, dtype, dformat, len(data or b""))
+        if self.max_clipboard_send_size > 0:
+            log.debug("perform clipboard limit checking - datasize - %d, %d", len(data), self.max_clipboard_send_size)
+            max_send_datalen = self.max_clipboard_send_size * 8 // get_format_size(dformat)
+            if len(data) > max_send_datalen:
+                olen = len(data)
+                data = data[:max_send_datalen]
+                log.warn("clipboard data copied out truncated because of clipboard policy %d to %d",
+                         olen, max_send_datalen)
```

Comment:
Just a note - not something to deal with now: we should probably do something smarter when handling strings.
Chopping the byte array might give an invalid UTF8 sequence.